cmd/rbac/main.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/controller-tools/pkg/generate/rbac"
+)
+
+func main() {
+	o := &rbac.ManifestOptions{}
+	cmd := &cobra.Command{
+		Use:   "rbac",
+		Short: "Generates RBAC manifests",
+		Long: `Generate RBAC manifests from the RBAC annotations in Go source files.
+Usage:
+# rbac generate [--name manager] [--input-dir input_dir] [--output-dir output_dir]
```

Comment:
I think the sub command "generate" here did not take effect. i.e. ”rbac [--name manager] [--input-dir input_dir] [--output-dir output_dir]“ can also work.

Maybe add it as a sub command explicitly

Reply:
Good pt. Created a new command `controller-gen` and added `rbac` as a subcommand. So here is the plan:

we will invoke `controller-gen all` using `go generate` directive from the generated project. `all` subcommand will trigger all the generators for a project.